### PR TITLE
[8/n] decouple quantization impl from vllm dependency - gguf srt

### PR DESCRIPTION
**Pull request by FlamingoPg**
## Motivation

Clean vllm dependency, as titled.

## Accuracy Tests

```bash
python3 /sgl-workspace/sglang/test/srt/test_gguf.py
[Test Method] test_models
/sgl-workspace/sglang/python/sglang/srt/sampling/sampling_params.py:17: DeprecationWarning: module 'sre_parse' is deprecated
  import sre_parse
`torch_dtype` is deprecated! Use `dtype` instead!
`torch_dtype` is deprecated! Use `dtype` instead!
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
[Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0
Capturing batches (bs=1 avail_mem=23.18 GB): 100%|████████████████████████████████| 2/2 [00:00<00:00,  2.82it/s]
.
----------------------------------------------------------------------
Ran 1 test in 66.670s

OK
```

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/srt/layers/quantization/gguf.py [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+        layer: torch.nn.Module,
+        dispatch_output: StandardDispatchOutput,
+    ) -> CombineInput:
+        assert self.fused_experts is None
+
+        from sglang.srt.layers.moe.token_dispatcher import StandardCombineInput
+
+        assert (
+            self.moe_runner_config.activation == "silu"
+        ), "Only SiLU activation is supported."
```
[the remark this answers is not in the document]
Comment (FlamingoPg) [reply python/sglang/srt/layers/quantization/gguf.py:526]: only support silu

**Review by merrymercy — APPROVED** [review]
